scripts/lua2dox.lua

```diff
@@ -158,6 +158,18 @@ local function removeCommentFromLine(line)
   return line:sub(1, pos_comment - 1), line:sub(pos_comment)
 end
 
+--- Add possible pattern variations to a type. This includes arrays, and
+--- optional types.
+---
+---@param type string
+---@return string
+local function addPossibleTypeCaptureGroups(type)
+  -- [%[%]]* : match any dimensional array
```

Comment:
Arrays weren't previously matched. This is why there are many diffs adding parentheses around array types.

Reply:
This will just match any consecutive strings with `[]`.e.g: `[]][][[][`, `[[[[[]`.